libs/vkd3d/memory.c

```diff
@@ -331,7 +331,9 @@ static void vkd3d_memory_allocation_free(const struct vkd3d_memory_allocation *a
 
     TRACE("allocation %p, device %p, allocator %p.\n", allocation, device, allocator);
 
+#ifdef VKD3D_ENABLE_DESCRIPTOR_QA
     vkd3d_descriptor_debug_unregister_cookie(device->descriptor_qa_global_info, allocation->resource.cookie);
```

Comment:
These should already be ifdeffed out to a ((void)0) expression.

Reply:
Resolved by only sorting out the structures in `vkd3d` headers.